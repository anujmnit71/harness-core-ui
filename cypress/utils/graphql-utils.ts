export const hasOperationName: (req: any, operationName: string) => boolean = (req, operationName) => {
  const { body } = req
  // eslint-disable-next-line no-prototype-builtins
  return body.hasOwnProperty('operationName') && body.operationName === operationName
}

export const aliasQuery: (req: any, operationName: string) => void = (req, operationName) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Query`
  }
}
