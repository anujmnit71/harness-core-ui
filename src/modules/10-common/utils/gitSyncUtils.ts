/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GitSyncConfig } from 'services/cd-ng'
import { folderPathName, yamlFileExtension } from './StringUtils'

export const getRepoDetailsByIndentifier = (identifier: string | undefined, repos: GitSyncConfig[]) => {
  return repos.find((repo: GitSyncConfig) => repo.identifier === identifier)
}

export const validateFilePath = (filePath: string): boolean => {
  if (!filePath.endsWith(yamlFileExtension)) {
    return false
  } else {
    const fullPath = filePath.slice(0, filePath.length - yamlFileExtension.length)
    const subPaths = fullPath.split('/')
    return subPaths.every((folderName: string) => {
      return folderName.length && folderName.match(folderPathName)?.length
    })
  }
}
