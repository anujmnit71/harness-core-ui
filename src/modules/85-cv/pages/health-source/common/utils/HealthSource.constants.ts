/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const HealthSourceFieldNames = {
  MANUAL_QUERY: 'isManualQuery',
  IDENTIFIER: 'identifier',
  METRIC_NAME: 'metricName',
  METRIC_DATA: 'metricData',
  GROUP_NAME: 'groupName',
  CONTINUOUS_VERIFICATION: 'continuousVerification',
  HEALTH_SCORE: 'healthScore',
  SLI: 'sli',
  RISK_CATEGORY: 'riskCategory',
  HIGHER_BASELINE_DEVIATION: 'higherBaselineDeviation',
  LOWER_BASELINE_DEVIATION: 'lowerBaselineDeviation'
}
