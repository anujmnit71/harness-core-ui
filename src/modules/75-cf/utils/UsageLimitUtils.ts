/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ReactNode } from 'react'
import { BannerType } from '@common/layouts/Constants'
import type { CheckFeatureReturn } from 'framework/featureStore/featureStoreUtil'
import { formatToCompactNumber } from '@cf/utils/CFUtils'
import type { UseStringsReturn } from 'framework/strings'

export const getBannerText = (
  getString: UseStringsReturn['getString'],
  monthlyActiveUsers: CheckFeatureReturn,
  additionalLicenseProps: Record<string, any>
): ReactNode => {
  const { isEnterpriseEdition, isFreeEdition, isTeamEdition } = additionalLicenseProps

  const clientMauUsageCount = Number(monthlyActiveUsers?.featureDetail?.count)
  const clientMauPlanLimit = Number(monthlyActiveUsers?.featureDetail?.limit)

  const clientMauUsagePercentage = Math.trunc((clientMauUsageCount / clientMauPlanLimit) * 100)
  const clientMauPlanLimitFormatted = formatToCompactNumber(clientMauPlanLimit)

  const showInfoBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 90 && clientMauUsagePercentage < 100
  const showWarningBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 100

  let message!: string
  let bannerType!: BannerType

  if (isFreeEdition) {
    if (!showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.freePlan.approachingLimit', {
            clientMauUsagePercentage,
            clientMauPlanLimitFormatted
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: getString('cf.planEnforcement.freePlan.upgradeRequired', {
          clientMauPlanLimitFormatted
        }),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  if (isTeamEdition || isEnterpriseEdition) {
    if (showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.teamEnterprisePlan.approachingLimit', {
            clientMauUsagePercentage
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: () => getString('cf.planEnforcement.teamEnterprisePlan.upgradeRequired'),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  return {
    message,
    bannerType
  }
}
