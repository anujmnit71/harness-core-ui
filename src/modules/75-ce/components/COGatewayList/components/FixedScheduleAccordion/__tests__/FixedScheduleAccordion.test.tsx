/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import * as lwServices from 'services/lw'
import type { Service } from 'services/lw'
import FixedScheduleAccordion from '../FixedScheduleAccordion'

const mockedStaticSchedules = [
  {
    id: 54,
    name: 'sched1',
    account_id: 'kmpySmUISimoRrJL6NL73w',
    created_by: 'lv0euRhKRCyiXWzS7pOg6g',
    details: {
      uptime: {
        period: {
          start: '2022-02-21T11:00:00+05:30',
          end: '2022-02-22T12:00:00+05:30'
        },
        days: {
          days: [0, 1, 2, 3, 4, 5, 6],
          all_day: true,
          start_time: null,
          end_time: null
        }
      },
      downtime: null,
      timezone: 'Asia/Calcutta'
    },
    description: 'Uptime, All day on Sun,Mon,Tue,Wed,Thu,Fri,Sat',
    resources: [
      {
        ID: '495',
        Type: 'autostop_rule'
      }
    ],
    created_at: '2022-02-21T17:38:56.395834Z',
    updated_at: '2022-02-21T17:38:56.395834Z'
  },
  {
    id: 55,
    name: 'sched2',
    account_id: 'kmpySmUISimoRrJL6NL73w',
    created_by: 'lv0euRhKRCyiXWzS7pOg6g',
    details: {
      uptime: null,
      downtime: {
        period: {
          start: '2022-02-23T00:00:00+05:30',
          end: '2036-04-24T00:00:00+05:30'
        },
        days: {
          days: [0, 1, 2, 3, 4, 5, 6],
          all_day: true,
          start_time: null,
          end_time: null
        }
      },
      timezone: 'Asia/Calcutta'
    },
    description: 'Downtime, All day on Sun,Mon,Tue,Wed,Thu,Fri,Sat',
    resources: [
      {
        ID: '495',
        Type: 'autostop_rule'
      }
    ],
    created_at: '2022-02-21T17:38:57.853004Z',
    updated_at: '2022-02-21T17:38:57.853004Z'
  }
]

const mockedServiceData: Service = {
  id: 123,
  name: '',
  status: 'stopped',
  cloud_account_id: 'cloud_id',
  kind: '',
  org_id: 'org_id'
}

jest.mock('services/lw', () => ({
  useListStaticSchedules: jest.fn().mockImplementation(() => ({
    data: { response: mockedStaticSchedules },
    loading: false,
    refetch: jest.fn()
  })),
  useDeleteStaticSchedule: jest.fn().mockImplementation(() => ({
    mutate: jest.fn().mockImplementation(() => Promise.resolve())
  })),
  useCreateStaticSchedules: jest.fn().mockImplementation(() => ({
    mutate: jest.fn().mockImplementation(() => Promise.resolve())
  })),
  useValidateStaticScheduleList: jest.fn().mockImplementation(() => ({
    mutate: jest.fn().mockImplementation(() => Promise.resolve())
  }))
}))

describe('Fixed schedules accordion tests', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })
  test('render schedule accordion with 2 schedules', () => {
    const { container } = render(
      <TestWrapper>
        <FixedScheduleAccordion service={mockedServiceData} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('clicking on edit icon should open fixed schedule form modal', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <FixedScheduleAccordion service={mockedServiceData} />
      </TestWrapper>
    )

    const btn = getByTestId('fixedSchedules-summary')
    expect(btn).toMatchSnapshot()
    fireEvent.click(btn)

    const editIcon = container.querySelector('span[data-icon="Edit"]')
    expect(editIcon).toBeDefined()
    act(() => {
      fireEvent.click(editIcon!)
    })
    const modal = document.querySelector('.bp3-dialog')
    expect(modal).toBeInTheDocument()
  })

  test('clicking on delete icon should open delete confirm modal', () => {
    const { container, getByTestId } = render(
      <TestWrapper>
        <FixedScheduleAccordion service={mockedServiceData} />
      </TestWrapper>
    )

    const btn = getByTestId('fixedSchedules-summary')
    expect(btn).toMatchSnapshot()
    fireEvent.click(btn)

    const deleteIcon = container.querySelector('span[data-icon="main-trash"]')
    expect(deleteIcon).toBeDefined()
    act(() => {
      fireEvent.click(deleteIcon!)
    })
    const modal = document.querySelector('.bp3-dialog')
    expect(modal).toBeInTheDocument()
  })

  test('render schedule accordion with 1 schedule', () => {
    jest.spyOn(lwServices, 'useListStaticSchedules').mockImplementation(
      () =>
        ({
          data: { response: [{ ...mockedStaticSchedules[0] }] },
          loading: false,
          refetch: jest.fn()
        } as any)
    )
    const { container } = render(
      <TestWrapper>
        <FixedScheduleAccordion service={mockedServiceData} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('render schedule accordion with 0 schedules', () => {
    jest.spyOn(lwServices, 'useListStaticSchedules').mockImplementation(
      () =>
        ({
          data: { response: null },
          loading: false,
          refetch: jest.fn()
        } as any)
    )
    const { container } = render(
      <TestWrapper>
        <FixedScheduleAccordion service={mockedServiceData} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('render schedule accordion while schedules are loading', () => {
    jest.spyOn(lwServices, 'useListStaticSchedules').mockImplementation(
      () =>
        ({
          data: { response: null },
          loading: true,
          refetch: jest.fn()
        } as any)
    )
    const { container } = render(
      <TestWrapper>
        <FixedScheduleAccordion service={mockedServiceData} />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
