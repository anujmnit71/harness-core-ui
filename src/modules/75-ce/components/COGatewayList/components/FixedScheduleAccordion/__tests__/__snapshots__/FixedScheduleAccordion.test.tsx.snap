// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixed schedules accordion tests clicking on delete icon should open delete confirm modal 1`] = `
<div
  class="Accordion--summary"
  data-testid="fixedSchedules-summary"
>
  <div
    class="Accordion--label"
  >
    2 ce.co.autoStoppingRule.configuration.step4.tabs.schedules.title
  </div>
  <div
    class="Accordion--chevron"
  />
</div>
`;

exports[`Fixed schedules accordion tests clicking on edit icon should open fixed schedule form modal 1`] = `
<div
  class="Accordion--summary"
  data-testid="fixedSchedules-summary"
>
  <div
    class="Accordion--label"
  >
    2 ce.co.autoStoppingRule.configuration.step4.tabs.schedules.title
  </div>
  <div
    class="Accordion--chevron"
  />
</div>
`;

exports[`Fixed schedules accordion tests render schedule accordion while schedules are loading 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--flex-alignItems-center"
  >
    <span
      class="bp3-icon StyledProps--main"
      data-icon="loading"
    >
      <svg
        height="16"
        viewBox="0 0 10 8"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          fill="currentColor"
          r="1"
          transform="translate(2 4)"
        >
          <animatetransform
            additive="sum"
            attributeName="transform"
            begin="0s"
            dur="1s"
            repeatCount="indefinite"
            type="scale"
            values="0; 1.25; 0"
          />
        </circle>
        <circle
          fill="currentColor"
          r="1"
          transform="translate(5 4)"
        >
          <animatetransform
            additive="sum"
            attributeName="transform"
            begin="0.2s"
            dur="1s"
            repeatCount="indefinite"
            type="scale"
            values="0; 1.25; 0"
          />
        </circle>
        <circle
          fill="currentColor"
          r="1"
          transform="translate(8 4)"
        >
          <animatetransform
            additive="sum"
            attributeName="transform"
            begin="0.4s"
            dur="1s"
            repeatCount="indefinite"
            type="scale"
            values="0; 1.25; 0"
          />
        </circle>
      </svg>
    </span>
  </div>
</div>
`;

exports[`Fixed schedules accordion tests render schedule accordion with 0 schedules 1`] = `<div />`;

exports[`Fixed schedules accordion tests render schedule accordion with 1 schedule 1`] = `
<div>
  <div
    class="Accordion--accordion serviceAccordionContainer"
  >
    <div
      class="Accordion--panel"
      data-open="false"
      data-testid="fixedSchedules-panel"
    >
      <div
        class="Accordion--summary"
        data-testid="fixedSchedules-summary"
      >
        <div
          class="Accordion--label"
        >
          1 ce.co.autoStoppingRule.configuration.step4.tabs.schedules.fixedSchedule
        </div>
        <div
          class="Accordion--chevron"
        />
      </div>
      <div
        class="bp3-collapse Accordion--collapse"
      >
        <div
          aria-hidden="false"
          class="bp3-collapse-body"
          style="transform: translateY(-undefinedpx);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Fixed schedules accordion tests render schedule accordion with 2 schedules 1`] = `
<div>
  <div
    class="Accordion--accordion serviceAccordionContainer"
  >
    <div
      class="Accordion--panel"
      data-open="false"
      data-testid="fixedSchedules-panel"
    >
      <div
        class="Accordion--summary"
        data-testid="fixedSchedules-summary"
      >
        <div
          class="Accordion--label"
        >
          2 ce.co.autoStoppingRule.configuration.step4.tabs.schedules.title
        </div>
        <div
          class="Accordion--chevron"
        />
      </div>
      <div
        class="bp3-collapse Accordion--collapse"
      >
        <div
          aria-hidden="false"
          class="bp3-collapse-body"
          style="transform: translateY(-undefinedpx);"
        />
      </div>
    </div>
  </div>
</div>
`;
